### PR TITLE
Update CatalogSource for Prometheus Operator deployment

### PR DESCRIPTION
**Pull request by leifmadsen**
Depended-by: https://github.com/infrawatch/documentation/pull/460

Please merge documentation PR once this merges.

### CODE REVIEW

**Comment by leifmadsen** [comment]
test

File: build/stf-run-ci/tasks/setup_base.yml [hunk 1]
```diff
@@ -103,13 +103,13 @@
       apiVersion: operators.coreos.com/v1alpha1
       kind: Subscription
       metadata:
-        name: prometheus
+        name: rhods-prometheus-operator
         namespace: "{{ namespace }}"
       spec:
         channel: beta
```
Comment (csibbitt) [inline build/stf-run-ci/tasks/setup_base.yml:109]: Is this the right channel?

Comment (leifmadsen) [reply build/stf-run-ci/tasks/setup_base.yml:109]: Yep, channel remained the same (I did look for it during docs updates)

File: build/stf-run-ci/tasks/setup_base.yml [hunk 2]
```diff
@@ -103,13 +103,13 @@
       apiVersion: operators.coreos.com/v1alpha1
       kind: Subscription
       metadata:
-        name: prometheus
+        name: rhods-prometheus-operator
```
[the remark this answers is not in the document]
Comment (csibbitt) [reply build/stf-run-ci/tasks/setup_base.yml:106]: So I left this subscription object name the same but changed the catalog and operator name. The subscription remained healthy but nothing happened even after the catalog polling interval. Likely the new CSV did not name our current one in the "replaces" field.

Then I deleted the existing CSV `oc delete csv prometheusoperator.0.56.3` and it took the operator pod along with it as expected. A minute or so later a CSV for rhods-prometheus-operator.4.10.0 was installed, the prom-op came back, my prometheus pod was updated, and smoke tests pass.

[the remark this answers is not in the document]
Comment (csibbitt) [reply build/stf-run-ci/tasks/setup_base.yml:106]: I guess I should try is as written too, to see if it's better some how?

[the remark this answers is not in the document]
Comment (csibbitt) [reply build/stf-run-ci/tasks/setup_base.yml:106]: It's not better. If I let a second subscription get created, then there are just two subscriptions and two csvs. I still need to delete the csv to stabilize things, but now I have a subscription to delete as well.

```
$ oc get sub
[...]
prometheus                             prometheus                             community-operators   beta
rhods-prometheus-operator              rhods-prometheus-operator              redhat-operators      beta

$ oc get csv
[...]
prometheusoperator.0.56.3                     Prometheus Operator                           0.56.3           prometheusoperator.0.47.0                     Succeeded
rhods-prometheus-operator.4.10.0              Prometheus Operator                           4.10.0                                                         Installing
```

[the remark this answers is not in the document]
Comment (csibbitt) [reply build/stf-run-ci/tasks/setup_base.yml:106]:
```suggestion
        name: prometheus
```

File: tests/infrared/README.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ to an STF instance all on one (large) baremetal machine.
 Once the deployment is complete, you can check prometheus for data, like so:
 
 ```shells
-$ PROM_HOST=$(oc get route prometheus -o jsonpath='{.spec.host}')
+$ PROM_HOST=$(oc get route default-prometheus-proxy -o jsonpath='{.spec.host}')
```
Comment (csibbitt) [inline tests/infrared/README.md:20]: This is not actually a result of the change, right? Just something you spotted?

**Review by csibbitt — CHANGES_REQUESTED** [review]

File: build/stf-run-ci/tasks/setup_base.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (csibbitt) [reply build/stf-run-ci/tasks/setup_base.yml:106]: I was thinking of suggesting we leave this name the same, but I've changed my mind.

I suppose this matter most during upgrades, not here in this file, but I think having code to delete the old sub + csv doesn't cost much more than code to delete just the csv, and if anything ever goes wrong, having each component clearly labelled helps with troubleshooting.

[the remark this answers is not in the document]
Comment (leifmadsen) [reply build/stf-run-ci/tasks/setup_base.yml:106]: OK I did some testing as well here, and confirmed that the PV created with the existing Prometheus remains. The Operator also works pretty nicely and makes sure the new Prometheus is running before terminating the old one, so there is no gap in the data. If the timing just happens to land in the wrong spot, I'd only expect a single scrape interval to be missed, worse case.

File: tests/infrared/README.md (hunk 3 above)
Comment (leifmadsen) [reply tests/infrared/README.md:20]: Yea just something spotted as I went through all the files.

**Review by csibbitt — APPROVED** [review]

**Review by csibbitt — APPROVED** [review]

**Comment by leifmadsen** [comment]
I'm going to leave this here for now, but the blockage for merging is the OCP 4.10 CI not having resources currently to deploy and run a test. I'm happy if someone wants to merge this down on Friday though as we have plenty of time to do manual testing in OCP 4.10.

**Comment by csibbitt** [comment]
The CI passed. I'm not sure what's up with these duplicate Build and Bundle checks, though? In any case, one copy of those passed too.

**Comment by leifmadsen** [comment]
Just noting we'll have to keep an eye out on longer term coordination of the Prometheus Operator from this channel. We may need to revert this at some point pending internal discussions about ownership and productization of this resource.

CC: @jparker-rh

**Pull request merged**